sources/Interop/Windows/winrt/wrl/client/ComPtr`1.Manual.cs

```diff
@@ -4,150 +4,401 @@
 // Original source is Copyright © Microsoft. All rights reserved.
 
 using System;
+using System.ComponentModel;
 using System.Diagnostics;
+using System.Runtime.CompilerServices;
 using static TerraFX.Interop.Windows;
 
 namespace TerraFX.Interop
 {
-    public unsafe partial struct ComPtr<T> : IDisposable
+    /// <summary>
+    /// A type that allows working with pointers to COM objects more securely.
+    /// </summary>
+    /// <typeparam name="T">The type to wrap in the current <see cref="ComPtr{T}"/> instance.</typeparam>
+    /// <remarks>
+    /// While this type is not marked as <see langword="ref"/> so that it can also be used in fields, make sure
+    /// to keep the reference counts properly tracked if you do store <see cref="ComPtr{T}"/> instances on the heap.
+    /// </remarks>
+    public unsafe struct ComPtr<T> : IDisposable
         where T : unmanaged
     {
-        private T* ptr_;
+        /// <summary>
+        /// The raw pointer to a COM object, if existing.
+        /// </summary>
+        private T* pointer;
 
+        /// <summary>
+        /// Creates a new <see cref="ComPtr{T}"/> instance from a raw pointer and increments the ref count.
+        /// </summary>
+        /// <param name="other">The raw pointer to wrap.</param>
         public ComPtr(T* other)
         {
-            ptr_ = other;
+            this.pointer = other;
+
             InternalAddRef();
         }
 
+        /// <summary>
+        /// Creates a new <see cref="ComPtr{T}"/> instance from a second one and increments the ref count.
+        /// </summary>
+        /// <param name="other">The other <see cref="ComPtr{T}"/> instance to copy.</param>
         public ComPtr(ComPtr<T> other)
         {
-            ptr_ = other.ptr_;
+            this.pointer = other.pointer;
+
             InternalAddRef();
         }
 
+        /// <summary>
+        /// Converts a raw pointer to a new <see cref="ComPtr{T}"/> instance and increments the ref count.
+        /// </summary>
+        /// <param name="other">The raw pointer to wrap.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static implicit operator ComPtr<T>(T* other)
         {
-            return new ComPtr<T>(other);
+            return new(other);
```

Comment:
I don't like this syntax and likely won't be using it in TerraFX. It hurts readability.

Reply:
Agreed it shouldn't always be used by default, I just figured it was fine in very small snippets like this.
Happy to revert this if you prefer though, done in b90b27e840a2b99fdaa552b9dd7f17385d0b056f 😊